be/src/http/action/stream_load.cpp

```diff
@@ -299,23 +309,44 @@ void StreamLoadAction::on_chunk_data(HttpRequest* req) {
     auto evbuf = evhttp_request_get_input_buffer(ev_req);
 
     int64_t start_read_data_time = MonotonicNanos();
-    while (evbuffer_get_length(evbuf) > 0) {
-        ByteBufferPtr bb = ByteBuffer::allocate(4096);
+
+    size_t len = 0;
+    while ((len = evbuffer_get_length(evbuf)) > 0) {
```

Comment:
If `len` is zero but `ctx->buffer` is nonempty, when will the buffer be flushed to the body sink?